src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final int figuresCount = 3;
+
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        FigureType randomFigureType = FigureType.values()[new Random().nextInt(figuresCount)];
```

Comment:
don't create random each time you want to use it - create it once, at the class-level field